### PR TITLE
suggestions for additional changes in 2.7.0

### PR DESCRIPTION
**Pull request by nevrome**
-  added column `Country_ISO` to janno file (if merged, this solves #40)
- added column `Library_Names`  to the janno file
- ~made `Capture_Type`, `UDG` and `Library_Built` to list columns to be ordered as `Libraries`~
- ~added option `other` to `UDG` in case there is another method of udg treatment not covered by the available options (is this a good idea?)~
- replaced the option `other` by `mixed` for the `Library_Built` janno column
- gave the seqSourceFile a proper extension: `.ssf`
- renamed the column `Poseidon_ID` to `poseidon_IDs` in the ssf
- added the columns `udg` and `library_built` to the ssf; they mirror `UDG` and `Library_Built` in the janno file, but without the option `mixed` (if merged, this solves #58)
- made janno+ssf-relationship a many-to-many relationship, by making the `poseidon_IDs` field a list column

### CODE REVIEW

**Review by stschiff — APPROVED** [review]
Looks good! Thanks

**Comment by 93Boy** [comment]
I have faced a problem while matching ENA data with Poseidon data. In some packages, I have found multiple Poseidon IDs like I10871.DG, I10871.SG, I10871_published What should be the procedure to map these into ENA data? If I add multiple Poseidon_IDs to such fields will it be harder to read by Trident or other tools?

File: sequencingSourceFile_columns.tsv [hunk 1]
```diff
@@ -19,4 +18,4 @@ fastq_aspera	The Aspera-link (URL) to the FASTQ-file(s). (Example: fasp.sra.ebi.
 fastq_bytes	The number of bytes of the FASTQ-file(s) in bytes	Integer	TRUE	FALSE	TRUE		0	Inf	FALSE	FALSE
 fastq_md5	The MD5 hash(es) of the FASTQ-file(s)	String	TRUE	FALSE	FALSE				FALSE	FALSE
 read_count	The number of reads	Integer	FALSE	FALSE	TRUE		0	Inf	FALSE	FALSE
-submitted_ftp	The URL(s) to the originally submitted file(s) before it got converted to FASTQ. This can sometimes be helpful for processing	String	TRUE	FALSE	FALSE				FALSE	FALSE
\ No newline at end of file
+submitted_ftp	The URL(s) to the originally submitted file(s) before it got converted to FASTQ. This can sometimes be helpful for processing	String	TRUE	FALSE	FALSE				FALSE	FALSE
```
Comment (stschiff) [inline sequencingSourceFile_columns.tsv:23]: what changed here? Newline?

Comment (nevrome) [reply sequencingSourceFile_columns.tsv:23]: Yeah - libreoffice must have added this. But I agree with it. There should be a newline at the end of the file.

**Comment by nevrome** [comment]
What's the role of `Nr_Libraries` with this new system? How shall I validate it? If we list the libraries explicitly then we don't need the number. But of course we have plenty of legacy data. Shall we just ignore it for now? @stschiff

**Comment by nevrome** [comment]
Ok - after the meeting today we have to modify this PR once more:

- The seqSourceFile should get a dedicated file extension: `.ssf`
- The file should also get a couple of new columns that mirror existing columns in the .janno file (but here per library). See https://github.com/poseidon-framework/poseidon2-schema/issues/58
- That means `Capture_Type`, `UDG` and `Library_Built` in the .janno file should NOT be list columns. That change has to be reverted
- That also means `Library_Built` should get a new option `mixed` (which it should have had anyway)

I'm not entirely clear if we should keep the new column `Libraries`. Probably it can be removed again, because it isn't really relevant on a sample/individual level. Opinions, @AyGhal, @TCLamnidis, @stschiff?

**Comment by stschiff** [comment]
> What's the role of `Nr_Libraries` with this new system? How shall I validate it? If we list the libraries explicitly then we don't need the number. But of course we have plenty of legacy data. Shall we just ignore it for now? @stschiff

See meeting today. What I'm not sure about is whether we should strictly validate that against the rows in the ssf file. I think we can't, since the granularity in the SSF file is not necessarily library, but could also be capture (see the Carlhoff example). So I think this could simply be a number, as before, and it's not validated against the SSF

**Comment by stschiff** [comment]
> I'm not entirely clear if we should keep the new column Libraries. Probably it can be removed again, because it isn't really relevant on a sample/individual level. Opinions, @AyGhal, @TCLamnidis, @stschiff?

Good question. Perhaps we simply can scrap that. I think it's confusing. See also my comment above.

**Comment by TCLamnidis** [comment]
I advocate for having a `Library_Names` (or `Libraries`) column generally. I think it can help with transparency. E.g. a second study publishes additional data for an existing individual, and we can have the `Library_Names` column to make it perfectly clear what is and is not part of the data. 
That said, I can see how this can/will become confusing with public data, where the ENA-provided library_names are sometimes libraries, sometimes captures, and sometimes everything that was produced for an individual in a bam that (hopefully) has RGs... 🤞 
I guess it's a nice-to-have generally, but might not be particularly useful in many cases.

Note: if this field is kept, I would like to name it `Library_Names` instead of `Libraries`. Firstly, I think that is more intuitive a name, and secondly that is what I've already called it in my own pipelines 😇

**Comment by nevrome** [comment]
I think I added all changes we discussed yesterday to the schema. I've also taken the freedom to make some minor decisions according to my preferences. The first comment above features an updated summary. Please check thoroughly, @stschiff, @TCLamnidis and @AyGhal.

I'm going back to my trident PR now to implement this, so that we can test it ASAP.

**Review by stschiff — APPROVED** [review]
Great. Approved!

**Pull request merged**